Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -404,8 +411,20 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                         }
                         else
                         {
-                            //TODO: Should probably check to see if the error is recoverable or transient.
-                            logger.Warn("Error occurred during request, retrying...", e);
+                            Exception innermostException = e;
+                            while (innermostException.InnerException != null && innermostException != innermostException.InnerException)

```

Comment:
maybe we could extract this logic to a method called GetInnerMostException()

Reply:
Done